### PR TITLE
#2340 Portfolio assets cleanup

### PR DESCRIPTION
**Pull request by drakkein**

### CODE REVIEW

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                  }
+                >
+                  <DocumentTemplateButton
+                    layout={EButtonLayout.SECONDARY}
+                    title={getDocumentTitles(isRetailEto)[document.documentType]}
+                  />
+                  <InlineIcon
+                    className={styles.downloadIcon}
+                    svgIcon={downloadIcon}
+                    height={"18px"}
```
Comment (Moejoe90) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: why not put height and width in the scss file?

File: app/modules/public-etos/sagas.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { compose, keyBy, map, omit } from "lodash/fp";
 import { delay } from "redux-saga";
 import { all, fork, put, race, select } from "redux-saga/effects";
 
+import BigNumber from "bignumber.js";
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: move above

File: app/modules/public-etos/sagas.ts [hunk 3]
```diff
@@ -335,6 +338,46 @@ function* downloadTemplateByType(
...
+  { contractsService }: TGlobalDependencies,
+  action: TActionFromCreator<typeof actions.publicEtos.loadTokensData>,
+): any {
+  const myAssets = yield select(selectMyAssets);
+
+  for (const eto of myAssets) {
+    const equityTokenAddress = eto.contract.equityTokenAddress;
+
+    const equityToken = yield contractsService.getEquityToken(equityTokenAddress);
+    const balance = yield equityToken.balanceOf(action.payload.walletAddress);
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: batch requests here (from 351 to 355), otherwise it will block ui longer than needed (use `all` from redux-saga)

File: app/components/portfolio/PortfolioAssetAction.tsx [hunk 4]
```diff
@@ -14,42 +18,60 @@ type TExternalProps = {
...
           <FormattedMessage id="portfolio.section.reserved-assets.claim-tokens" />
         </Button>
       );
-    // case ETOStateOnChain.Refund:
-    //   return (
-    //     <Button layout="simple" svgIcon={arrowIcon} iconPosition="icon-after">
-    //       <FormattedMessage id="portfolio.section.reserved-assets.refund" />
-    //     </Button>
-    //   );
+    case EETOStateOnChain.Refund:
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioAssetAction.tsx]: since it's not developed and now you uncommented it  I would add a `disabled` here

File: app/modules/public-etos/sagas.ts [hunk 5]
```diff
@@ -335,6 +338,46 @@ function* downloadTemplateByType(
...
+    const tokenController = yield equityToken.tokenController;
+
+    const controllerGovernance = yield contractsService.getControllerGovernance(tokenController);
+    const [
+      totalCompanyShares,
+      companyValuationEurUlps,
+      ,
+    ] = yield controllerGovernance.shareholderInformation();
+
+    const tokenPrice = new BigNumber(
```
Comment (desfero) [inline app/modules/public-etos/sagas.ts]: Probably we should not convert it back to `BigNumber` as there is a force from @AlexeyTuboltsev to use plain types inside `store`

File: app/modules/reducer.ts [hunk 6]
```diff
@@ -63,4 +64,5 @@ export const appReducers = {
   investmentFlow: investmentFlowReducer,
   videoModal: videoModalReducer,
   personProfileModal: personProfileModalReducer,
+  portfolioDownloadAgreementsModal: portfolioDownloadAgreementsModalReducer,
```
Comment (desfero) [inline app/modules/reducer.ts]: @drakkein for future reference it's better to use `GenericModal` as this complicates reducer a lot with unneeded states

File: app/modules/public-etos/actions.ts [hunk 7]
```diff
@@ -49,4 +49,14 @@ export const etoActions = {
     (previewCode: string, data: IEtoContractData) => ({ previewCode, data }),
   ),
   setEtoWidgetError: createActionFactory("PUBLIC_ETOS_SET_ETO_WIDGET_ERROR"),
+  loadTokensData: createActionFactory("PORTFOLIO_LOAD_TOKENS_DATA", (walletAddress: string) => ({
```
Comment (desfero) [inline app/modules/public-etos/actions.ts]: Move to `public actions` as this is not state mutation

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+        neuValue: multiplyBigNumbers([myNeuBalance, neuPrice]),
+        myAssets: selectMyAssetsWithTokenData(state)!,
+      };
+    },
+    dispatchToProps: dispatch => ({
+      showDownloadAgreementModal: (etoId: string, isRetailEto: boolean) => {
+        dispatch(actions.portfolio.showDownloadAgreementModal(etoId, isRetailEto));
+      },
+      loadTokensData: (walletAddress: string) => {
+        dispatch(actions.publicEtos.loadTokensData(walletAddress));
```
Comment (desfero) [inline app/components/portfolio/PortfolioMyAssets.tsx]: btw, why this `loadTokensData` is not inside `InvestorTickets` as this is strictly related to porfolio only? Or not?

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  selectDownloadAgreementModalIsRetailEto,
+  selectDownloadAgrementModalIsOpen,
+} from "../../portfolio/selectors";
+import { EButtonLayout } from "../../shared/buttons";
+import { DocumentTemplateButton } from "../../shared/DocumentLink";
+import { SectionHeader } from "../../shared/SectionHeader";
+import { ModalComponentBody } from "../ModalComponentBody";
+
+import * as downloadIcon from "../../../assets/img/inline_icons/download.svg";
+import { InlineIcon } from "../../shared/InlineIcon";
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: move above

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            }, eto.templates)}
+            <p className="mb-0" />
+          </section>
+        )}
+      </ModalComponentBody>
+    </Modal>
+  );
+};
+
+const DownloadTokenAgreementModal = compose<any, any>(
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: `any`, `any`?

Comment (desfero) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: If you do not expect any props probably you should use `{}` so it's clear for reader

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+  appConnect<IStateProps, IDispatchProps>({
+    stateToProps: state => {
+      const etoId = selectDownloadAgreementModalEtoId(state);
+      const eto = etoId ? selectEtoWithCompanyAndContractById(state, etoId) : undefined;
+      return {
+        isOpen: etoId ? selectDownloadAgrementModalIsOpen(state) : false,
+        isRetailEto: etoId ? selectDownloadAgreementModalIsRetailEto(state)! : false,
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: do we need to `!` at the end?

File: app/components/portfolio/PortfolioAssetAction.tsx [hunk 12]
```diff
@@ -14,42 +18,60 @@ type TExternalProps = {
...
+          svgIcon={arrowRight}
+          innerClassName={"text-uppercase"}
+        >
+          <FormattedMessage id="portfolio.section.reserved-assets.refund" />
+        </Button>
+      );
     case EETOStateOnChain.Signing:
       return (
-        <Button layout={EButtonLayout.SECONDARY} disabled>
+        <Button layout={EButtonLayout.SIMPLE} disabled innerClassName={"text-uppercase"}>
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioAssetAction.tsx]: why did you remove the `disabled` button this is also an unfinished feature

File: app/lib/web3/ContractsService.ts [hunk 13]
```diff
@@ -95,6 +99,26 @@ export class ContractsService {
     this.etoCommitmentCache[etoId] = contract;
     return contract;
   }
+
+  async getEquityToken(equityTokenAddress: string): Promise<IEquityToken> {
+    if (this.equityTokensCache[equityTokenAddress])
+      return this.equityTokensCache[equityTokenAddress];
+
+    const web3 = this.web3Manager.internalWeb3Adapter.web3;
```
Comment (AlexeyTuboltsev) [inline app/lib/web3/ContractsService.ts]: hey, this statement repeats 4 times! (lines 49, 97, 107, 117). Maybe just add it as a private member to this class?

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+                  <InlineIcon
+                    className={styles.downloadIcon}
+                    svgIcon={downloadIcon}
+                    height={"18px"}
+                    width={"18px"}
+                  />
+                </div>
+              ) : null;
+            }, eto.templates)}
+            <p className="mb-0" />
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: Do we need empty `p`?

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+  &:nth-of-type(2n) {
+    background-color: transparent;
+  }
+
+  .download-icon {
+    margin-left: auto;
+    margin-right: 15px;
+
+    svg {
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss]: do we need this if you provide size as a prop?

File: app/components/shared/NewTable.tsx [hunk 16]
```diff
@@ -63,21 +76,28 @@ const NewTable: React.FunctionComponent<TProps> = ({
...
           <thead className={styles.header}>
             <tr>
-              {titles.map((title, index) => (
-                <th className={styles.cell} key={index}>
-                  {title}
-                </th>
-              ))}
+              {titles.map((title, index) => {
+                return typeof title === "object" && title && "width" in title ? (
+                  <th className={styles.cell} key={index} style={{ width: title.width }}>
```
Comment (AlexeyTuboltsev) [inline app/components/shared/NewTable.tsx]: better would be
return title && typeof title === "object" && "width" in title ?

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+      width: 18px;
+      height: 18px;
+    }
+
+    path {
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss]: maybe it's possible to set `color` to `.download-icon` as not you break coupling because you assume that component inside will have `path` node

File: app/components/portfolio/PortfolioAssetAction.tsx [hunk 18]
```diff
@@ -14,42 +18,60 @@ type TExternalProps = {
...
       );
-    // case ETOStateOnChain.Refund:
-    //   return (
-    //     <Button layout="simple" svgIcon={arrowIcon} iconPosition="icon-after">
-    //       <FormattedMessage id="portfolio.section.reserved-assets.refund" />
-    //     </Button>
-    //   );
+    case EETOStateOnChain.Refund:
+      return (
+        <Button
```
Comment (desfero) [inline app/components/portfolio/PortfolioAssetAction.tsx]: But refund is not ready yet?

Comment (desfero) [reply app/components/portfolio/PortfolioAssetAction.tsx]: It even don't have any `onClick` attached so it's useless button to show

File: app/components/portfolio/PortfolioLayout.tsx [hunk 19]
```diff
@@ -89,224 +64,8 @@ const PortfolioLayout: React.FunctionComponent<TPortfolioLayoutProps> = ({
...
-                      ) : null;
-                    }, templates)}
-                  </>
-                </NewTableRow>
-              );
-            },
-          )}
-        </NewTable>
-      </Col>
-    </Row>
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioLayout.tsx]: :+1:

File: app/components/portfolio/PortfolioAssetAction.tsx [hunk 20]
```diff
@@ -14,42 +18,60 @@ type TExternalProps = {
...
+          layout={EButtonLayout.SIMPLE}
```
Comment (desfero) [inline app/components/portfolio/PortfolioAssetAction.tsx]: Simple button should be only used when you want to build your own custom button as it doen't have any behavior attached for e.g. `:hover`, why we can not stick with `SECONDARY`?

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+                value={neuValue}
+                currency={ECurrency.EUR}
+                currencySymbol={ECurrencySymbol.SYMBOL}
+              />
+              <Money
+                value={neuPrice}
+                currency={ECurrency.EUR}
+                currencySymbol={ECurrencySymbol.SYMBOL}
+              />
+              <ButtonLink
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioMyAssets.tsx]: There is a new component that got merged lately for blank targets can you please use it

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+                innerClassName={"text-uppercase"}
+              >
+                <FormattedMessage id="portfolio.section.my-assets.download-agreements" />
+              </ButtonLink>
+            </NewTableRow>
+          ) : null}
+
+          {myAssets &&
+            myAssets
+              .filter(v => v.tokenData)
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioMyAssets.tsx]: I'm not sure but be a bit cautious here as `tokenData` are all bigNumbers. So I don't see much use for checking if they will be available.

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+
+const PortfolioMyAssets = compose<TComponentProps, IExternalProps>(
+  appConnect<IStateProps, IDispatchProps>({
+    stateToProps: state => {
+      const neuPrice = selectNeuPriceEur(state);
+      const myNeuBalance = selectNeuBalance(state.wallet);
+
+      return {
+        myNeuBalance,
+        neuPrice: new BigNumber(neuPrice).toFixed(8),
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioMyAssets.tsx]: this looks like a good use case for a selector

File: app/components/shared/NewTable.tsx [hunk 24]
```diff
@@ -63,21 +76,28 @@ const NewTable: React.FunctionComponent<TProps> = ({
...
+                    {title.title}
```
Comment (AlexeyTuboltsev) [inline app/components/shared/NewTable.tsx]: now we have either a title (string/TTranslated smth) or an object with many fields unrelated to title. This all gets too convoluted. I suggest to leave the titles to be simply an array of titles (e.g. strings/translations) and if you need to force width on some columns, do it separately. Like add another field to NewTable -- Columns, an array of column widths  (like[null, 5px, null, 20px]).

or, maybe it's easier
convert the titles to on object everywhere, call it columnSettings or smth like this, and pack all this stuff in this object. Like ColumnSettings{title?:string|translation, width?:number, customCSS?:string } etc

File: app/components/portfolio/PortfolioMyAssets.tsx [hunk 25]
```diff
@@ -0,0 +1,184 @@
+import * as cn from "classnames";
+import * as React from "react";
+import { FormattedMessage } from "react-intl-phraseapp";
+import { Col, Row } from "reactstrap";
+import { compose, lifecycle } from "recompose";
+
+import BigNumber from "bignumber.js";
```
Comment (Moejoe90) [inline app/components/portfolio/PortfolioMyAssets.tsx]: import order please

File: app/modules/public-etos/sagas.ts [hunk 26]
```diff
@@ -335,6 +338,46 @@ function* downloadTemplateByType(
...
+export function* loadTokensData(
+  { contractsService }: TGlobalDependencies,
+  action: TActionFromCreator<typeof actions.publicEtos.loadTokensData>,
+): any {
+  const myAssets = yield select(selectMyAssets);
+
+  for (const eto of myAssets) {
+    const equityTokenAddress = eto.contract.equityTokenAddress;
+
+    const equityToken = yield contractsService.getEquityToken(equityTokenAddress);
```
Comment (Moejoe90) [inline app/modules/public-etos/sagas.ts]: This can all be packed into one `Promise.all`
have a look at this
https://github.com/Neufund/platform-frontend/blob/71f8594dba75274b312fcc0862e0b1fc1d57704b/app/modules/wallet-selector/ledger-wizard/sagas.ts#L55
Or this example from a WIP branch
https://github.com/Neufund/platform-frontend/pull/2355/files#diff-453914c9822223caa684238d4e1cd5f2R88

File: app/components/portfolio/reducer.ts [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+      return {
+        isOpen: true,
+        etoId: action.payload.etoId,
+        isRetailEto: action.payload.isRetailEto,
+      };
+    }
+    case actions.portfolio.hideDownloadAgreementModal.getType(): {
+      return {
+        isOpen: false,
+      };
```
Comment (AlexeyTuboltsev) [inline app/components/portfolio/reducer.ts]: maybe return initial state then? Or remove etoId and isRetailEto from initial state, for consistency's sake

File: app/components/appRoutes.ts [hunk 28]
```diff
@@ -26,5 +26,5 @@ export const appRoutes = {
 
   etoRegister: "/eto/registration",
 
-  portfolio: "/portfolio",
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/appRoutes.ts]: woah, that was Webstorms refactoring tool

File: app/components/portfolio/PortfolioReservedAssets.tsx [hunk 29]
```diff
@@ -0,0 +1,136 @@
...
+
+    <Row>
+      <Col className={styles.reservedAssets}>
+        <NewTable
+          keepRhythm={true}
+          panelStyle={styles.panel}
+          placeholder={
+            <FormattedMessage id="portfolio.section.reserved-assets.table.header.placeholder" />
+          }
+          titles={[
```
Comment (AlexeyTuboltsev) [inline app/components/portfolio/PortfolioReservedAssets.tsx]: see comment below (app/components/shared/NewTable.tsx). This is too messy. Need to separate formatting (forcing width etc) from content (titles)

File: app/components/portfolio/PortfolioLayout.tsx [hunk 30]
```diff
@@ -89,224 +64,8 @@ const PortfolioLayout: React.FunctionComponent<TPortfolioLayoutProps> = ({
...
+    <PortfolioReservedAssets pendingAssets={pendingAssets} />
+    <PortfolioMyAssets isRetailEto={isRetailEto} walletAddress={walletAddress} />
```
Comment (AlexeyTuboltsev) [inline app/components/portfolio/PortfolioLayout.tsx]: 👍

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 31]
```diff
@@ -0,0 +1,146 @@
...
+              <FormattedMessage id="portfolio.section.my-assets.modal.header" />
+            </SectionHeader>
+            {map((document: IEtoDocument) => {
+              return [EEtoDocumentType.SIGNED_INVESTMENT_AND_SHAREHOLDER_AGREEMENT].includes(
+                document.documentType,
+              ) ? (
+                <div
+                  key={document.documentType}
+                  className={styles.document}
+                  onClick={() =>
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: onClick on div is not correct

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss [hunk 32]
```diff
@@ -0,0 +1,25 @@
+.document {
+  display: flex;
+  align-items: center;
+  height: 46px;
+  background-color: $neu-pale-blue;
+  cursor: pointer;
```
Comment (desfero) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss]: cursor: pointer should not be added to not focusable elements

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx [hunk 33]
```diff
@@ -0,0 +1,146 @@
...
+  onClose,
+  generateTemplateByEtoId,
+  downloadDocument,
+  eto,
+  isRetailEto,
+}) => {
+  return (
+    <Modal isOpen={isOpen} toggle={onClose}>
+      <ModalComponentBody onClose={onClose}>
+        {eto && (
```
Comment (AlexeyTuboltsev) [inline app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: what happens if there's no eto? Maybe add a branch with loading state or some error message?

File: app/modules/public-etos/sagas.ts (hunk 5 above)
Comment (Moejoe90) [reply app/modules/public-etos/sagas.ts]: :+1: 
For sure we need to only put plain types inside store 
We already have a case where date is in the store 
https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state

File: app/components/portfolio/PortfolioLayout.module.scss [hunk 34]
```diff
@@ -7,14 +7,30 @@
...
+  text-transform: uppercase;
+  white-space: nowrap;
+}
+
 .document-link {
   margin-bottom: px-to-rem(16px);
   text-align: left;
 }
+
+.reserved-assets,
```
Comment (desfero) [inline app/components/portfolio/PortfolioLayout.module.scss]: We should not change styles for tables only in portfolio as this is inconsistent and will lead to some weird UX. If design team wants to change table then it should change all of them

File: app/components/portfolio/PortfolioMyAssets.tsx (hunk 8 above)
Comment (rudolfix) [reply app/components/portfolio/PortfolioMyAssets.tsx]: equity token data is not part of InvestorTickets. you can easily have equity token that you bought on secondary market and never invested in it via ETO. so you can have a token in MY ASSETS that you do not have and never had in PENDING ASSETS. you could just got it via `transfer` function from other investor.
So having separate state for Tokens Data is good

**Comment by rudolfix** [comment]
I've checked price calculations in MY ASSETS and it's good. thanks @drakkein

File: app/modules/reducer.ts (hunk 6 above)
Comment (drakkein) [reply app/modules/reducer.ts]: Lets leave it this way, I'll create ticket to refactor this

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx (hunk 10 above)
Comment (drakkein) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: 👍

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss (hunk 17 above)
Comment (drakkein) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.module.scss]: it wont work :(

File: app/components/portfolio/PortfolioAssetAction.tsx (hunk 18 above)
Comment (drakkein) [reply app/components/portfolio/PortfolioAssetAction.tsx]: Lets disable it

File: app/components/portfolio/PortfolioAssetAction.tsx (hunk 20 above)
Comment (drakkein) [reply app/components/portfolio/PortfolioAssetAction.tsx]: ok, but there is need to inject additional styles via `innerClassName` with `padding: 0 !important;` to achieve correct look

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx (hunk 31 above)
Comment (drakkein) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: changed to be anchor

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx (hunk 1 above)
Comment (drakkein) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: It's set anyway to override inline svgs size, I'll leave it only in scss

File: app/components/shared/NewTable.tsx (hunk 24 above)
Comment (drakkein) [reply app/components/shared/NewTable.tsx]: Let's make additional issue for `NewTable` improvements

File: app/components/portfolio/reducer.ts (hunk 27 above)
Comment (drakkein) [reply app/components/portfolio/reducer.ts]: initialState is best option i suppose

File: app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx (hunk 33 above)
Comment (drakkein) [reply app/components/modals/download-token-agreements-modal/DownloadTokenAgreementModal.tsx]: this condition is only for initial state, there should be no case when eto is missing and modal is open

File: app/components/portfolio/PortfolioReservedAssets.tsx (hunk 29 above)
Comment (drakkein) [reply app/components/portfolio/PortfolioReservedAssets.tsx]: Like previously, let's make additional issue for `NewTable`
